### ISSUE TITLE
Lack of functionality QDesktopServices

### ISSUE THREAD

**Issue by avttrue** [QuasarApp/CQtDeployer#817]
Build config: KDE Neon, Qt 6.7.1
Test config KDE Neon (Wayland), KDE Neon (X11)

Greetings. 
I found such a problem. 
- There are no problems when running a built application from the build directory :) 
- When running application from the "DistributionKit" directory, the `QDesktopServices::openURL(const QUrl &url)` functionality does not work. 

How I build:
`cqtdeployer -bin [MyApp] -extraLibs "libb2,libxcb-cursor,libvdpau_va_gl" -enablePlugins "wayland-graphics-integration-client,wayland-graphics-integration-server" -disablePlugins "platformthemes,styles,wayland-decoration-client" -qmake ~/Qt/6.7.1/gcc_64/bin/qmake -verbose 3 >&verboselog.txt`

As I understand it, something does not get into the "lib" directory. Because I copied the "plugins" directory completely from Qt. Can you tell me how to fix this problem?

Additional files:
[lib.txt](https://github.com/user-attachments/files/15863764/lib.txt)
[plugins.txt](https://github.com/user-attachments/files/15863766/plugins.txt)
[ldd.txt](https://github.com/user-attachments/files/15863768/ldd.txt)
[verboselog.txt](https://github.com/user-attachments/files/15863776/verboselog.txt)

**Comment by avttrue** [comment]
I'll clarify. The functionality of working with files and directories disappears. The web links are functioning.

**Comment by EndrII** [comment]
If it's trouble of the distribution (plugins or some else) the Qt should print error message on the application log after invoke openURL function.

Check please what happened after invoke openURL function.

**Comment by avttrue** [comment]
> If it's trouble of the distribution (plugins or some else) the Qt should print error message on the application log after invoke openURL function.
> 
> Check please what happened after invoke openURL function.

Nothing, only my logging. "qt.multimedia.ffmpeg.libsymbolsresolver: Couldn't load VAAPI library" at start - does not affect the work of the application.

[log.txt](https://github.com/user-attachments/files/15864242/log.txt)

**Comment by avttrue** [comment]
By the way. 
Just observations. 
- It works from the build directory. 
- Through Qt Creator - does **not** work. 
- Through DistributionKit/karass.sh - does **not** work. 
- Through Distribution Kit/bin/karass - it works. 

Could this have something to do with the rights in the console? 
Although, I use everything by default. It's about opening files/directories.
No. I have now checked the launch from the console / mc - what worked, it works there.

**Comment by EndrII** [comment]
> By the way. Just observations.
> 
>     * It works from the build directory.
> 
>     * Through Qt Creator - does **not** work.
> 
>     * Through DistributionKit/karass.sh - does **not** work.
> 
>     * Through Distribution Kit/bin/karass - it works.
> 
> 
> Could this have something to do with the rights in the console? Although, I use everything by default. It's about opening files/directories. No. I have now checked the launch from the console / mc - what worked, it works there.

It's very weird behavior, because qt Desktop located in the GUI library that load immediately with your app.

**Comment by avttrue** [comment]
> t's very weird behavior, because qt Desktop located in the GUI library that load immediately with your app.

I build a test application, the same story.
[Архив.zip](https://github.com/user-attachments/files/15866637/default.zip)

**Comment by avttrue** [comment]
I can upload a [DistributionKit](https://disk.yandex.ru/d/9ghT6ZhSNfGYkg)

**Comment by EndrII** [comment]
> I can upload a [DistributionKit](https://disk.yandex.ru/d/9ghT6ZhSNfGYkg)

i run your test application. its works fine for me. 
OS Ubuntu 24.04 (Wayland)

Probably trouble in your OS. I can't help you with this issue, because I can't reproduce it. 

One way testing for you: 
if you install Qt from Distribution repository, remove them. And try again. If you use Qt- corp prebuilt framework, remove them and rebuild your app with qt from repository. 
 
Just remove Qt for your OS 


We already have a one incident with system-depended functions of Qt. 
see https://github.com/QuasarApp/CQtDeployer/issues/808#issuecomment-2069508602

**Comment by avttrue** [comment]
> > I can upload a [DistributionKit](https://disk.yandex.ru/d/9ghT6ZhSNfGYkg)
> 
> i run your test application. its works fine for me. OS Ubuntu 24.04 (Wayland)
> 
> Probably trouble in your OS. I can't help you with this issue, because I can't reproduce it.

Thank you!
I have now checked the Distribution Kit on KDE Neon Plasma 5 (LTS) - the problem is not reproduced i.e. everything works correctly. 
Initially, I was testing on Plasma 6, this is not the LTS version. 
I attribute this problem to a glitch in my non-LTS Plasma. 
Yes, I installed Qt with the standard installer. 

P.S. I looked at how directories are opened in the Qt Creator sources  - my respect, no some QDesktopServices, etc.
